js/controller/todoFilterController.js

```diff
@@ -0,0 +1,102 @@
+'use strcit';
+
+import { api } from '../api/api.js';
+import { todoAppView } from '../view/todoAppView.js';
+import { teamStore } from '../store/teamStore.js';
+import { memberStore } from '../store/memberStore.js';
+import { elementValidator } from '../utils/validator.js';
+import { MESSAGE } from '../constant/message.js';
+import { FILTER_TYPE } from '../constant/constants.js';
+
+export default class TodoFilterController {
+  constructor() {
+    todoAppView.$todoappListContainer.addEventListener(
+      'click',
+      this.onClickFilter
+    );
+  }
+
+  onClickFilter = ({ target }) => {
+    if (
+      elementValidator.isNotFilterBtn(target) &&
+      elementValidator.isNotClearBtn(target)
+    )
+      return;
+    if (elementValidator.isFilterBtn(target)) {
+      this.changeByFilter(target);
+      return;
+    }
+    if (elementValidator.isClearBtn(target)) {
+      this.clearAllItems(target);
+      return;
+    }
+  };
+
+  getMemberId(target) {
+    return target.closest('.todoapp-container').dataset.id;
+  }
+
+  getMember(target) {
+    const memberId = this.getMemberId(target);
+    return memberStore.findMember(memberId);
+  }
+
+  async clearAllItems(target) {
+    if (!confirm(MESSAGE.DELETE_ALL_ITEM)) return;
+    const teamId = teamStore.getCurrentTeam()._id;
+    const memberId = this.getMemberId(target);
+    await api.deleteTodoItems(teamId, memberId);
+    const team = await api.getTeam(teamId);
+    memberStore.setMembers(team.members);
+    todoAppView.renderKanban(memberStore.getMembers());
+  }
+
+  getCurrentOption(target) {
+    const classList = target.classList;
+    if (classList.contains(FILTER_TYPE.ALL)) {
+      return FILTER_TYPE.ALL;
+    }
+    if (classList.contains(FILTER_TYPE.ACTIVE)) {
+      return FILTER_TYPE.ACTIVE;
+    }
+    if (classList.contains(FILTER_TYPE.PRIORITY)) {
+      return FILTER_TYPE.PRIORITY;
+    }
+    if (classList.contains(FILTER_TYPE.COMPLETED)) {
+      return FILTER_TYPE.COMPLETED;
+    }
```

Comment:
```suggestion
const filterTypes = {
  [FILTER_TYPE.ALL]: FILTER_TYPE.ALL,
  [FILTER_TYPE.ACTIVE]: FILTER_TYPE.ACTIVE
}

Object.keys(filterTypes).find((type) => {
  return classList.contains(type) && filterTypes[type]
})
```

현재 구조는 필터의 타입이 길어질수록 조건문이 계속 늘어날텐데요. 필터 타입을 객체로 관리하고 위와 같이 구현해본다면 어떨까요~?